src/base_home.h

```diff
@@ -0,0 +1,142 @@
+#define BASE_HOME
+
+#include "submap.h"
+#include <unordered_map>
+#include "ammo.h"
+#include "string_id.h"
+
+struct submap;
+struct tripoint;
+class ammunition_type;
+
+using furn_id = int_id<furn_t>;
+using ammotype_id = string_id<ammunition_type>;
+
+/**
+ * Flags unique to a base.
+ */
+enum base_area_flag
+{
+    BASE_IN,   /**Inside of actual base*/
+    BASE_GND,  /**On base grounds. i.e. between perimeter fence and building*/
+    BASE_WALL, /**Terrain seperating BASE_IN and BASE_GND*/
+    BASE_EDGE, /**outer edge of base area*/
+    NO_NPC,    /**NPCs stay out*/
+};
+
+//furniture flags in json
+//BASE_CORE -> Is the very heart and soul of a base.
+//BASE_CMD# -> Level provided by control core.
+//CMD_SYS -> Acts as a data conduit for the command system.  If adjacent or forms a path to BASE_CORE, allows placement of aux systems.
+//AUX_SYS -> Is an auxiliary system. Must be built adjacent to BASE_CORE flag or CMD_SYS flag linked to a BASE_CORE flag.
+//Connectable -> Possible to install network module
+//NETWORKED -> Network enabled; able to form connection between command and remote system.
+
+/**
+ * Stores capacities of command system and the auxiliary systems present.
+ */
+struct command_sys
+{
+    bool network = false;  /**Is the command system wifi enabled? (cmd core must be networked)*/
+    bool auto_conn = true; /**Automaticaly connect to remote systems if able*/
+    int max_conn = 0;      /**Network capacity (in generic units).*/
+    int max_cpu;           /**Maximum processing power (in generic units.*/
+
+    std::unordered_map<furn_id, int> aux_sys;    /**Maps <auxiliary system type, # in command system>*/
+    std::unordered_map<furn_id, int> remote_sys; /**Maps <remote system type, # connected to command system*/
+};
+
+class base_home
+{
+
+    base(submap &base_map, const tripoint &coreloc);
+
+  private:
+    //########PRIVATE##########//
+    //#BASE CORE AND LOCATIONS#//
+    //#########################//
+
+    //data members
+    int base_level; /**Level of command core. Determins max # of aux sys and, use of certain features.*/
+    command_sys cmd_sys;
+
+    std::list<base_area_flag> baflag[SEEX][SEEY]; /**additional ter and fur flags specific to bases.*/
+    std::list<npc_based *> freeloaders;           /**list containing references to the NPCs staying at the base.*/
+
+    std::list<tripoint> bunks;        /**list containing locations of sleeping spots in base.*/
+    std::list<tripoint> storage_open; /**list containing locations of unclaimed/designated storage furnature.*/
+
+    //functions
+    void define_base_area(const submap &base_map, const tripoint &coreloc); //populates baflag[][]
+
+    //########PRIVATE##########//
+    //#ITEMS AND STASHED STUFF#//
+    //#########################//
+
+    //data members
+    std::unordered_map<ammotype_id, int> gun_count;  /**stores <ammo_type, #guns_that_use_them>.*/
+    std::unordered_map<ammotype_id, int> ammo_count; /**stores <ammo_types, amount>.*/
```

Comment:
This is much harder to keep updated than you think. Leave it for much later.

Reply:
won't actually be useful until based npcs are actually added so sure.